react/src/utils/Context.js

```diff
@@ -0,0 +1,124 @@
+// -------------------------------------------------------------------------------------------------
+// Context (location) related functions.
+// I.e. Locations of client (React UI), server (React API), authentication (Auth0).
+// -------------------------------------------------------------------------------------------------
+
+const _CLIENT_BASE_PATH  = "/api/react";
+const _SERVER_BASE_PATH  = "/api/reactapi";
+const _SERVER_LOCAL_PORT = 8000
+
+// -------------------------------------------------------------------------------------------------
+// Client (React UI) context info.
+// -------------------------------------------------------------------------------------------------
+
+function GetClientOrigin() {
+    return window.location.origin;
+}
+
+function GetClientDomain() {
+    return window.location.hostname;
+}
+
+function GetClientBasePath() {
+    return _CLIENT_BASE_PATH;
+}
+
+function GetClientBaseUrl() {
+    return GetClientOrigin() + GetClientBasePath();
+}
+
+function GetClientCurrentPath() {
+    return window.location.pathname;
+}
+
+function IsLocalClient() {
+    const origin = GetClientOrigin();
+    return origin.startsWith("http://localhost:") || origin.startsWith("http://127.0.0.1:");
+}
+
+// -------------------------------------------------------------------------------------------------
+// Server (React API) context info.
+// -------------------------------------------------------------------------------------------------
+
+function GetServerOrigin() {
+    //
+    // N.B. If the client (React UI) is running locally then assume the server (React API) is as well.
+    //
+    return IsLocalClient() ? "http://localhost:" + _SERVER_LOCAL_PORT : window.location.origin;
```

Comment:
Added support for in this code for running https locally, and added comments about this in general (at the top), but I haven't tested https locally, and not sure it would work WRT authentication because the Auth0 configuration would need to include an https version of localhost, which I kinda doubt it does.